src/main/java/backend/UpdateController.java

```diff
@@ -61,6 +62,15 @@ public void processAndRefresh(List<FilterPanel> filterPanels) {
                     if (toUpdate.isEmpty()) {
                         // If no issues requiring metadata update, just run the filter and sort.
                         logic.updateUI(processFilters(filterExprs));
+                        ArrayList<CompletableFuture<Boolean>> stub = new ArrayList<>();
+                        Futures.sequence(stub)
+                                .thenRun(() -> {
+                                    if (isResumeTimer) {
+                                        logic.refreshTimer.resume();
+                                    }
+                                })
+                                .thenCompose(n -> logic.getRateLimitResetTime())
+                                .thenApply(logic::updateRateLimits);
```

Comment:
We have some mechanism in the back-end for such scenario so that data is updated consistently although there might be concurrent refresh. Concurrent download of data is allowed since they do not affect each other, but the data update is packaged into mutually exclusive operations and put in the blocking queue for each repository and is safe.

It's still inefficient however. But I think such problems should better be optimised by managing operations in the queues instead of handling in an adhoc way through the timer (e.g. removing similar consecutive operations in the queue). Anyway, this can be discussed another time :-)